### PR TITLE
psa_skip API

### PR DESCRIPTION
**Pull request by romkuz01**
Implementation of
size_t psa_skip(psa_handle_t msg_handle, uint32_t invec_idx, size_t num_bytes)
API as described here: https://github.com/ARMmbed/PSA-IPC-doc#psa_skip

3 new tests for this API:
- skip_zero
- skip_some
- skip_more_than_left

### CODE REVIEW

**Comment by mikisch81** [comment]
@romkuz01 the PR diff is wrong, need to rebase or something

File: spm/spm_server.c [hunk 1]
```diff
@@ -200,6 +200,32 @@ size_t psa_read(psa_handle_t msg_handle, uint32_t iovec_idx, void *buf, size_t n
     return num_bytes;
 }
 
+size_t psa_skip(psa_handle_t msg_handle, uint32_t invec_idx, size_t num_bytes)
```
Comment (mikisch81) [inline spm/spm_server.c]: This function is 90% similar to `psa_read()`, the only differences are:
- Signature
- buffer check
- Copying to buffer

Maybe, in order to reduce code size (As Gilad told in his lecture which you will see at GEC 😃) we should have a static function which will do both functionalities?

Comment (mikisch81) [reply spm/spm_server.c]: Something like this:
```c
size_t psa_read(psa_handle_t msg_handle, uint32_t invec_idx, void *buf, size_t num_bytes)
{
    if (!is_buffer_accessible(buf, num_bytes)) {
        SPM_PANIC("buffer is inaccessible\n");
    }
    return internal_read(msg_handle, invec_idx, buf, num_bytes);
}

size_t psa_skip(psa_handle_t msg_handle, uint32_t invec_idx, size_t num_bytes)
{
    return internal_read(msg_handle, invec_idx, NULL, num_bytes);
}

static size_t internal_read(psa_handle_t msg_handle, uint32_t invec_idx, void *buf, size_t num_bytes)
{
    if (invec_idx >= PSA_MAX_INVEC_LEN) {
        SPM_PANIC("Invalid invec_idx\n");
    }

    active_msg_t *active_msg = NULL;
    spm_msg_handle_get_mem(msg_handle, &active_msg);

    SPM_ASSERT((active_msg->type > PSA_IPC_MSG_TYPE_INVALID) &&
               (active_msg->type <= PSA_IPC_MSG_TYPE_MAX));

    PARTITION_STATE_ASSERT(active_msg->channel->dst_sec_func->partition, PARTITION_STATE_ACTIVE);

    iovec_t *active_iovec = &active_msg->in_vec[invec_idx];

    if (num_bytes > active_iovec->iov_len) {
        num_bytes = active_iovec->iov_len;
    }

    if (num_bytes > 0) {
        if (buf) {
            memcpy(buf, active_iovec->iov_base, num_bytes);
        }
        active_iovec->iov_base = (void *)((uint8_t*)active_iovec->iov_base + num_bytes);
        active_iovec->iov_len -= num_bytes;
    }

    return num_bytes;
```

**Comment by orenc17A** [comment]
replaced by PR #5

**Pull request closed without merging**